src/feeder/JSONNode.cpp

```diff
@@ -0,0 +1,194 @@
+#include "JSONNode.h"
+
+// JSONNode default constructor
+JSONNode::JSONNode() {}
+
+// JSONNode copy constructor
+JSONNode::JSONNode(const JSONNode& t) {
+  node_id = t.node_id;
+  node_name = t.node_name;
+  node_type = t.node_type;
+  is_cpu_op = t.is_cpu_op;
+  runtime = t.runtime;
+  data_deps = t.data_deps;
+  dep_unresolved_parent_ids_json = t.dep_unresolved_parent_ids_json;
+  children_vec_json = t.children_vec_json;
+  children_set_json = t.children_set_json;
+  /* 	COMM_SEND_NODE = 5,
```

Comment:
Let's use the autogenerated C++ files and variables in the code, rather than hardcoding values directly.

Reply:
The enum if defined in the autogenerated protobuf header and I did not want to add protobuf dependency to JSON node. I can replicate the enumerated data structure in the JSON node header and use those variables for readability.